### PR TITLE
Admin api

### PR DESCRIPTION
**Pull request by Nyutonyka**
An error appears on startup: Parameter 0 of constructor in com.store.service.impl.ProductAdminServiceImpl required a bean of type 'com.store.repository.ProductAdminRepository' that could not be found.

### CODE REVIEW

File: src/main/java/com/store/controller/ProductAdminController.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    private final ProductAdminServiceImpl productAdminService;
+
+    @Autowired
+    public ProductAdminController(ProductAdminServiceImpl productAdminService) {
+        this.productAdminService = productAdminService;
+    }
+
+    @PostMapping
+    public ResponseEntity<Product> addProduct(@RequestBody ProductAdminDto productAdminDto) {
+        Product product = productAdminService.addProduct(productAdminDto);
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/controller/ProductAdminController.java:29]: It is a good approach to isolate DB entity from any external client.
FE knows only ProductAdminDto. Product - it is our internal entity class. If FE sends ProductAdminDto, please return then ResponseEntity<ProductAdminDto>. And always return only ProductAdminDto, not entity class. You need to create mapping from Product to ProductAdminDto.

File: src/main/java/com/store/service/impl/ProductAdminServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    @Override
+    public Product addProduct(ProductAdminDto productAdminDto) {
+        Product product = new Product();
+        product.setArticle(productAdminDto.getArticle());
+        product.setName(productAdminDto.getName());
+        product.setPrice(productAdminDto.getPrice());
+        product.setCategory(productAdminDto.getCategory());
+        product.setDescription(productAdminDto.getDescription());
+        product.setQuantity(productAdminDto.getQuantity());
+        return productAdminRepository.save(product);
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/service/impl/ProductAdminServiceImpl.java:25]: To make this code more clear you can use @Builder from lombok. This code also will work, but if you wish, you can change it.

File: src/main/java/com/store/controller/ProductAdminController.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * @version 0.0.1
+ */
+
+
+@RestController
+@RequestMapping("/admin/products")
+public class ProductAdminController {
+
+    private final ProductAdminServiceImpl productAdminService;
+
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/controller/ProductAdminController.java:21]: If you created an interface it makes sense to inject interface here

File: src/main/java/com/store/service/ProductAdminService.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.store.service;
+
+import com.store.dto.ProductAdminDto;
+import com.store.entity.Product;
+
+public interface ProductAdminService {
+
+    Product addProduct(ProductAdminDto productAdminDto);
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/service/ProductAdminService.java:8]: This interface is not needed here, but if you want you can leave it

**Comment by MariaTarasiuk** [comment]
> 

I don't understand why you added exclude here: @SpringBootApplication(exclude = {DataSourceAutoConfiguration.class })
Try to delete exclude, maybe it will fix a problem.

**Comment by MariaTarasiuk** [comment]
As I see, this PR is in main. But as I remember we decided main - only for tested tasks. Dev - for development and testing

File: src/main/java/com/store/controller/ProductAdminController.java (hunk 3 above)
Comment (Nyutonyka) [reply src/main/java/com/store/controller/ProductAdminController.java:21]: Corrected

File: src/main/java/com/store/controller/ProductAdminController.java (hunk 1 above)
Comment (Nyutonyka) [reply src/main/java/com/store/controller/ProductAdminController.java:29]: Here I did not understand. Everything is correct or not. If not, can you tell me how to fix it?

**Comment by Nyutonyka** [comment]
@SpringBootApplication(exclude = {DataSourceAutoConfiguration.class }) I added it because there was an error without BD connection. Now I deleted this annotation and all the work

Regarding the branch, can we try to leave it as we initially talked?

File: src/main/java/com/store/service/ProductAdminService.java (hunk 4 above)
Comment (Nyutonyka) [reply src/main/java/com/store/service/ProductAdminService.java:8]: I will leave the interfaces

File: src/main/java/com/store/controller/ProductAdminController.java (hunk 1 above)
Comment (MariaTarasiuk) [reply src/main/java/com/store/controller/ProductAdminController.java:22]: Your method in Controller should return ResponseEntity<ProductAdminDto>. What means addProduct() in productAdminService should return ProductAdminDto. Query to DB returns Product, so you need to convert Product to ProductAdminDto in service.

Comment (Nyutonyka) [reply src/main/java/com/store/controller/ProductAdminController.java:22]: I reworked it a bit and I don't return ResponseEntity. We will return the id of the created product. Everything works, the product is added to the database

**Comment by Nyutonyka** [comment]
Added validation, checked all work

File: src/main/java/com/store/dto/ProductAdminDto.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import lombok.experimental.Accessors;
+
+import java.math.BigDecimal;
+
+@Data
+@Accessors(chain = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class ProductAdminDto {
+    @NotNull(message = "Article is required")
```
Comment (MariaTarasiuk) [inline src/main/java/com/store/dto/ProductAdminDto.java:19]: you need to add one more field here with Product id. It is a primary key in DB.

**Pull request closed without merging**